semgrep-core/src/core/ast/Map_AST.ml

```diff
@@ -1099,3 +1099,21 @@ let (mk_visitor : visitor_in -> visitor_out) =
     }
   in
   all_functions
+
+(*****************************************************************************)
+(* Fix token locations *)
+(*****************************************************************************)
+
+let mk_fix_token_locations fix =
+  mk_visitor
+    {
+      default_visitor with
+      kstmt =
+        (fun (k, _) s ->
+          k
+            {
+              s with
+              s_range = Common.map_opt (fun (x, y) -> (fix x, fix y)) s.s_range;
```

Comment:
good call, I would have forgotten that one.

Reply:
I forgot too, but the tests reminded me about it!